### PR TITLE
[template-files][tools-public] don't use jq in check-dynamic-macros-android.sh

### PR DESCRIPTION
**Pull request by dsokal**
# Why

https://github.com/expo/turtle/issues/200 

We started using `jq` in the `tools-public/check-dynamic-macros-android.sh` script. We run this script in Turtle/Turtle CLI. `jq` is not a standard thing installed on users' computers. In this PR I'm getting rid of `jq` from the script.

# How

We decided to use `jq` because we changed the format of `template-files/{android,ios}-paths.json` files so they had two keys - `paths` and `generateOnly`. Each of the keys contained a mapping between template and target files. The former contained the files that were both generated with `et PLATFORM-generate-dynamic-macros` and then checked for existence in `check-dynamic-macros-android.sh` (in the case of Android, I'm not sure why we need it for iOS - I assume the reason was to have the same format of files). The latter contained the files that were only generated (and not checked).

I changed the format so we have two files now:
- `template-files/{android,ios}-paths.json` - contains all the files to generate
- `template-files/{android,ios}-paths.check-ignore` (any suggestions for a better name?) - contains the list of files that don't have to be check for the existence 

I also reimplemented scripts using those files.

This has to be backported to SDK37.

# Test Plan

- I'm not sure how to test that properly, however...
- I ran `./check-dynamic-macros-android.sh` in `tools-public` and it succeeded without errors.
- I ran `et ios-generate-dynamic-macros` and then `et android-generate-dynamic-macros` and I made sure both still work fine.
- Any ideas about what else is worth checking?

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -45,6 +45,9 @@ xcuserdata
 .project
 .settings
 
+# VSCode
+.history/
+
```
Comment (dsokal) [inline .gitignore:50]: Should I move this to a separate commit?

**Review by sjchmiela — APPROVED** [review]
I think this should be ok. 👍

**Review by tsapeta — APPROVED** [review]

**Review by mczernek — APPROVED** [review]
LGTM
